elements/dash-tab-panel/dash-tab-panel.js

```diff
@@ -4,174 +4,217 @@
-    // it was a keypress inside the a panel or on empty space. Nothing to do.
-    if (event.target.getAttribute('aria-role') !== 'tab') return;
-    // TODO: Why are we skipping on alt?
-    if (event.altKey) return;
-
-    let newIdx;
-    let newTab;
-    const tabs = Array.from(this.querySelectorAll('[aria-role=tab]'));
-
-    // The switch-case will determine which tab should be marked as active
-    // depending on the key that was pressed.
-    switch (event.keyCode) {
-      // When up or left is pressed, the previous tab is selected. If the
-      // first tab is reached, the element loops around to the last tab.
-      case 37: // 'ArrowLeft':
-      case 38: // 'ArrowUp':
-        newIdx =
-          tabs.findIndex(panel => panel.classList.contains('selected')) - 1;
-        newTab = tabs[(newIdx + tabs.length) % tabs.length];
-        break;
-
-      // Same goes for right and down, but in the other direction.
-      case 39: // 'ArrowRight':
-      case 40: // 'ArrowDown':
-        newIdx =
-          tabs.findIndex(panel => panel.classList.contains('selected')) + 1;
-        newTab = tabs[(newIdx + tabs.length) % tabs.length];
-        break;
-
-      // The home button always selects the first tab.
-      case 36: // 'Home':
-        newTab = tabs[0];
-        break;
-
-      // And end always selects the last one.
-      case 35: // 'End':
-        newTab = tabs[tabs.length - 1];
-        break;
-      // Any other key press is ignored and passed back to the browser.
-      default:
-        return;
+    /**
+     * `disconnectedCallback` removes the event listeners that
+     * `connectedCallback` added.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('click', this._onClick);
     }
-    // Select the new tab, that has been determined in the switch-case.
-    this._selectTab(newTab);
-    // The browser might have some native functionality bound to the arrow
-    // keys. Prevent the browser from taking any actions.
-    event.preventDefault();
-  }
 
-  /**
-   * `_handleClick` handles clicks inside the tab panel.
-   */
-  _handleClick(event) {
-    // If the click was not targeted on a tab element itself,
-    // it was a click inside the a panel or on empty space. Nothing to do.
-    if (!event.target.getAttribute('aria-role', 'tab')) return;
-    // If it was on a tab element, though, select that tab.
-    this._selectTab(event.target);
+    /**
+     * `selectTab` marks the given tab as selected.
+     * Additionally, it unhides the panel corresponding to the given tab.
+     */
+    _selectTab(newTab) {
+      // Deselect all tabs and hide all panels.
+      this.reset();
+
+      const panels = Array.from(this.querySelectorAll('[role=tabpanel]'));
+      // Get the panel that corresponds to the given tab.
+      const newPanelId = newTab.getAttribute('aria-controls');
+      const newPanel = panels.find(panel => panel.id === newPanelId);
```

Comment:
The above two lines are duplicated at 77,78. Might consider turning them into a helper function